### PR TITLE
feat(ui): if zui tag is not found for download, it now builds locally…

### PR DESCRIPTION
**Pull request by aokirisaki**
… from the commit

**What type of PR is this?**

**Which issue does this PR fix**:
#1271 

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -410,8 +410,20 @@ ui:
 		rm -rf ./pkg/extensions/build;\
 		cp -R $$tdir/zui/build ./pkg/extensions/;\
 	else\
-		curl -fsSL https://github.com/project-zot/zui/releases/download/$(ZUI_VERSION)/zui.tgz -o zui.tgz;\
-		tar xvzf zui.tgz -C ./pkg/extensions/;\
-		rm zui.tgz;\
+		curl --fail --head https://github.com/project-zot/zui/releases/download/$(ZUI_VERSION)/zui.tgz;\
+		if [ $$? -ne 0 ]; then\
+  			git clone https://github.com/project-zot/zui.git;\
```
Comment (andaaron) [inline Makefile:415]: Let's clone in a temporary folder. See line 403

Comment (rchincha) [reply Makefile:415]: Also git clone **that** version - $(ZUI_VERSION)?

Comment (aokirisaki) [reply Makefile]: I have made the changes

**Review by peusebiu — APPROVED** [review]
lgtm

**Review by andaaron — APPROVED** [review]

**Pull request merged**